### PR TITLE
Add a generic volume_histogram function.

### LINKED ISSUE #37: add script: dos+vdx -> DVHs
one command to generate all DVH plots 
optional: export all DVH data to .dat files instead.

### LINKED ISSUE #38: add script: .dosemlet.dos+vdx -> LVHs
same as issue #37 but using LET cubes.

### LINKED ISSUE #360: add generic volume_histogram function
just a suggestion

### PR DESCRIPTION
**Pull request by nbassler**
closes #360 
closes #37 
closes #38

### CODE REVIEW

**Comment by nbassler** [comment]
@grzanka a few questions
- wonder if this is a proper way how to calculate e.g. DVHs... It is rather different from the implemented solution in the DosCube and LETCube classes.
- I like when methods have a verb in them, because this distinguishes them from attributes. In that case, this method could be renamed to `make_volume_histogram()` or `get_volume_historgram()` What do you think?

File: pytrip/util.py [hunk 1]
```diff
@@ -43,3 +43,33 @@ def f(x):
...
+    :params cube: a data cube of any shape, e.g. Dos.cube
+    :params voi: optional voi where histogramming will happen.
+    :returns [x],[y]: coordinates ready for plotting. Dose (or LET) along x, Normalized volume along y in %.
+
+    If VOI is not given, it will calculate the histogram for the entire dose cube.
+
+    Providing voi will slow down this function a lot, so if in a loop, it is recommended to do masking
+    i.e. only provide Dos.cube[mask] instead.
+    """
+    import numpy as np
```
Comment (grzanka) [inline pytrip/util.py]: This should go to top level of the module.

File: pytrip/util.py [hunk 2]
```diff
@@ -43,3 +43,33 @@ def f(x):
...
+
+    if voi is None:
+        mask = np.ones(cube.shape, dtype=bool)
```
Comment (grzanka) [inline pytrip/util.py]: This is an unnecessary memory overhead. `mask = None` should also work:

```
>>> import numpy as np
>>> x = np.array([[1,2,3],[4,5,6]])
>>> x
array([[1, 2, 3],
       [4, 5, 6]])
>>> mask = None
>>> x[mask]
array([[[1, 2, 3],
        [4, 5, 6]]])
```

File: pytrip/util.py [hunk 3]
```diff
@@ -43,3 +43,33 @@ def f(x):
...
+    else:
+        vcube = voi.get_voi_cube()
+        mask = (vcube == 1000)
+
+    _xrange = (0.0, cube.max()*1.1)
+    _hist, x = np.histogram(cube[mask], bins=256, range=_xrange)
```
Comment (grzanka) [inline pytrip/util.py]: What about making `bins` a parameter of `volume_histogram` with default value 256 ?

File: pytrip/util.py [hunk 4]
```diff
@@ -43,3 +43,33 @@ def f(x):
...
+    _fhist = np.flip(_hist, axis=0)  # reverse historgram, so first element is for highest dose
+    _fhist = np.cumsum(_fhist)
+    _hist = np.flip(_fhist, axis=0)  # flip back again to normal representation
+    y = 100.0 * _hist / _hist[0]
+    return x[:-1], y  # TODO: think about what a proper representation would be
```
Comment (grzanka) [inline pytrip/util.py]: Why not the same as `np.histogram`, mainly values + bins edges (of len(values) + 1)?

Comment (nbassler) [inline pytrip/util.py]: what?

File: pytrip/util.py [hunk 5]
```diff
@@ -43,3 +45,35 @@ def f(x):
...
+        mask = None
+    else:
+        vcube = voi.get_voi_cube()
+        mask = (vcube.cube == 1000)
+
+    _xrange = (0.0, cube.max()*1.1)
+    _hist, x = np.histogram(cube[mask], bins=bins, range=_xrange)
+    _fhist = np.flip(_hist, axis=0)  # reverse historgram, so first element is for highest dose
+    _fhist = np.cumsum(_fhist)
+    _hist = np.flip(_fhist, axis=0)  # flip back again to normal representation
```
Comment (grzanka) [inline pytrip/util.py]: For python older than 3.4 we need to use numpy < 1.12, see:
https://github.com/pytrip/pytrip/blob/master/requirements.txt

Older version of numpy do not have `flip` method.
Consider using sth like:
```
x = x[::-1]
```

This is the reason travis job fails: https://travis-ci.org/pytrip/pytrip/jobs/226449935

**Comment by nbassler** [comment]
@grzanka should we get rid of the old DVH and LVH routines? May simplify code and less maintenance.

**Comment by nbassler** [comment]
@grzanka you may want take a look at the new dvhplot.py

**Comment by nbassler** [comment]
@grzanka do you have some suggestion how to fix the missing DISPLAY environment case? Setting the matplotlib backend results that the plot is not shown anymore.

**Comment by grzanka** [comment]
@nbassler What you propose in a505b021235f9f5d4d49d1acb52365f9617b81ff looks correct. The idea of separation between pytrip and pytripgui means that the first one is responsible only for logic (or work as a backend) and shouldn't pop-up with any display to the user. It should be solely console application. Generating files is OK, but displaying them not. 

Maybe we should encourage users to use jupyter-notebook and display the images there.

**Pull request merged**